### DISCUSSION TITLE
Download reports by annotation type

### DISCUSSION THREAD

**Discussion by aburfeid** [biigle/core#572] [Ideas]
Hello,

We were wondering if it would be possible to download the reports by annotation type. Right now we work like this: Annotation of the number of diatoms required (diatom count), and then creation of "?" annotations which are written into a separate excel sheet to confirm biodiversity (biodiversity survey). We download the data using the Abundance counts. In an ideal world we could have the "?"-taxa that we annotate identified within BIIGLE. 

We have thought also about the possibility of using separate label trees, but, as commented in our discussions #484 and as you stated #228 taxonomy keeps changing and would be very prone to create two separate and quite distinct label trees.

If there was a possibility to download the report by "type of annotation", that would make the accountability and leave the repeatability intact, if we use e.g. square bounding boxes for the counts and circles for the biodiversity survey. If it was possible to even double down, separating label trees and types of annotation, this would be ideal, but we can work around it if the "type of annotation" existed.

Thank you in advance.

Best wishes,

Andrea

P.S. I searched for "reports" and "annotation types" and did not find a similar request.

**Comment by mzur** [comment]
Sorry but I didn't understand your workflow, yet. Could you please provide a specific example or link to an image where I can have a look?

**Reply by aburfeid** [reply]
Example of a slide worked through with this method: https://biigle.de/images/2826132/annotations?r=4329&x=22166&y=22158
You can see that we use the lawnmowermode to arrive to the 500 valves needed for the count, after that use the random sampling to look for other, not yet identified diatom species (which affect the biodiversity but are not part of the count). It is standard practice in "traditional microscopy" to document the biodiversity, even if it is not part of the count. I know I am explaining it quite complicatedly, but as it is a "gold standard" practice, we wanted to implement it within BIIGLE, if possible.

Right now we use point-annotations with the "?" label for the "not-yet-recorded species" in the slide, as if we identify them, they get added to our count, which is not the purpose. If it were possible to download the abundance report not only by user or label tree, but by annotation tool type, that would make it possible for us to identify the species within BIIGLE and having a "count" and a "biodiversity survey" tab.

**Reply by aburfeid** [reply]
And just to make it a bit clearer:

1. Identification of diatom-valves to the species label using the lawnmower mode. Counting up to the stipulated number (usually 400-500 valves per sample)
2. Change to random sampling mode to look for not yet identified taxa, to make a biodiversity survey of the site. Currently we use the "?" annotation using the point annotation tool, as we have no possibility to keep the identifications out of the valve count, so the identifications are kept in a separate Excel file.*
3. When all samples are done, download the Abundance count, we currently use the label tree subdivision.

If in 2* and 3 there was a possibility that we could download the report by type of label, that would make it possible for us, to have the diatom counts and the biodiversity survey separate from one another.

**Reply by mzur** [reply]
Thanks, now I got it. So the different annotation shape really is a workaround. You actually want a report of all annotations without the "?" label and all annotations with the "?" label (and possibly additional labels). However, this would require a complicated "query builder" for reports which we don't plan to implement.

So I'd say the different annotation shapes are a *good* workaround :wink: I've added this as a feature idea (https://github.com/biigle/reports/issues/102).

An alternative workaround could be annotation sessions. You can already restrict a report to a specific annotation session. If you create the "count" annotations during one session and the "?" annotations during another, you can generate two separate reports. The only downside is that annotation sessions are valid for whole days. So you have to finish creating the "count" annotations on one day and then have to wait until the next day to start the "?" annotations.

**Reply by aburfeid** [reply]
Great, thank you. We also had thought about the annotation sessions, but as we prefer to do the biodiversity survey directly after the count, it would not really be practicable.

Thank you so much for the quick reply :)
